src/main/java/net/greatstart/controllers/UserController.java

```diff
@@ -1,39 +1,78 @@
 package net.greatstart.controllers;
 
+import net.greatstart.dto.DtoUserProfile;
 import net.greatstart.model.User;
+import net.greatstart.services.UserConverterService;
 import net.greatstart.services.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
 
-import java.util.List;
+import javax.validation.Valid;
+import java.security.Principal;
 
 @Controller
+@RequestMapping("/user")
 public class UserController {
+    private static final String SHOW_PROFILE = "user/showProfile";
+    private static final String EDIT_PROFILE = "user/editProfile";
+    private static final String ERROR_PROFILE = "redirect:/";
 
     private UserService userService;
+    private UserConverterService userConverter;
 
     @Autowired
-    public UserController(UserService userService) {
+    public UserController(UserService userService, UserConverterService userConverterService) {
         this.userService = userService;
+        this.userConverter = userConverterService;
     }
 
-    public boolean createUser(User user) {
-        return userService.createUser(user);
+    // TODO error pages (no such user etc.); enable admin user to edit profiles
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)